src/test/scala/com/tsunderebug/scolortesting/otf/OTFByteAllocatorTest.scala

```diff
@@ -0,0 +1,40 @@
+package com.tsunderebug.scolor.otf
+
+import com.tsunderebug.scolor.Offset
+import com.tsunderebug.scolor.otf.types.OTFOffset32
+
+import spire.math.{UByte, UInt}
+
+import org.scalatest.{FlatSpec, PrivateMethodTester}
+
+class OTFByteAllocatorTest extends FlatSpec with PrivateMethodTester {
+  
+  /* use PrivateMethodTester to access private values in class */
+  val nextAvailableOffset = PrivateMethod[Offset]('nextAvailableOffset)
+
+  "allocate" should "move the next available offset correctly" in {
+    val otfByteAllocator = new OTFByteAllocator(OpenTypeFont(Nil))    
+    otfByteAllocator.allocate(UInt(10))
+
+    val nextOffset = otfByteAllocator invokePrivate nextAvailableOffset()
+    assertResult(UInt(10))(nextOffset.position)
+  }
+
+  it should "return an offset describing the next open space these bytes can fit" in {
+    val otfByteAllocator = new OTFByteAllocator(OpenTypeFont(Nil))    
+    
+    /* first we expect 0 */
+    assertResult(OTFOffset32(0)) {
```

Comment:
In the test above I used `position` to get the UInt from spire instead of relying on the offset type, but here I use `OTFOffset32` directly. 

Do you have a preference on this? I'll make this code use one style, but just wanted to get your thoughts

Reply:
Test both, they should both work

Reply:
As in, split both of the tests you have into a test using case class comparison and a class using .position, as one compares how the case class in the allocator is holding up and the other shows whether or not the position of an Offset is working correctly.

Reply:
@TsundereBug
I think my latest change does what you want if you want to check it out